### ISSUE TITLE
symfony/http-client - Stop working with profiler

### ISSUE THREAD

**Issue by jdupouyes** [symfony/symfony#49219]
### Symfony version(s) affected

5.4.20

### Description

With symfony/http-client 5.4.20 HTTP requests no longer appear in the HTTP Client tab of the Symfony profiler.

### How to reproduce

Create a controller with an action with an HttpClientInterface parameter.
In the action code, make a call to a valid URL with the client:
```php
<?php

namespace App\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
use Symfony\Component\HttpFoundation\Response;
use Symfony\Component\Mailer\MailerInterface;
use Symfony\Component\Routing\Annotation\Route;
use Symfony\Contracts\HttpClient\HttpClientInterface;

class TestController extends AbstractController
{
    /**
     * @Route("/test")
     */
    public function test(HttpClientInterface $httpClient): Response
    {
        $httpClient->request('GET','http://google.com');
        return $this->render('index.html.twig');
    }
}
```

With the project in dev mode with debug activated, call the route and go to the profiler:
- With symfony/http-client 5.4.19 an entry is added to the HTTP Client tab of the profiler

![image](https://user-images.githubusercontent.com/91727581/216594395-715bb803-94d6-4d55-b88a-dcc29229a528.png)

- With symfony/http-client 5.4.20 the HTTP Client tab of the profiler is empty

![image](https://user-images.githubusercontent.com/91727581/216594546-37fcb694-ab13-4710-9063-b57b0d985e82.png)

**Comment by schodemeiss** [comment]
This seems to have appeared after the CVE fixes.

**Comment by bigfoot90** [comment]
Same for **6.2.5** version too.

I can see the requests tracked by commenting out the IF-ELSE in
 `vendor/symfony/web-profiler-bundle/Resources/views/Collector/http_client.html.twig` file
![screencapture-localhost-profiler-5401a9-2023-02-03-21_11_46](https://user-images.githubusercontent.com/4598274/216704744-2a7c9ce4-6415-458b-a030-420677352f6d.jpg)

**Comment by fpaterno** [comment]
Problem has appeared with this PR : https://github.com/symfony/symfony/pull/49104/commits/3cb1d70f215f52ff281a4935eef01bf2775f6edd

I feel `reset()` should be called first (at the beginnig of the method)

**Comment by OskarStark** [comment]
cc @nicolas-grekas

**Comment by dmamchyts** [comment]
What about the same fix into 6.2 and other versions?

![image](https://user-images.githubusercontent.com/102530135/218378972-8d67adb1-c68f-43dc-b825-bf71c924cbd9.png)

**Comment by fancyweb** [comment]
@dmamchyts The patch you screenshotted was already merged in 6.2.

https://github.com/symfony/symfony/pull/49301 is a new patch that has been merged in 5.4 and will be merged up to 6.2. You have to wait until the next release cycle which should be at the end of the month.
